system-variables: add @@global.tidb_accelerate_user_creation_update (#20290)
pingcap/docs: system-variables.md

- Default value: (system dependent)
- This variable shows the system time zone from when TiDB was first bootstrapped. See also [`time_zone`](#time_zone).

### tidb_accelerate_user_creation_update <span class="version-mark">New in v9.0.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Boolean
- Default value: `OFF`

> - When the number of SQL users in the system is very large (for example 100,000), operations such as creating users, modifying passwords, and setting permissions might experience significant performance degradation. This occurs because each user update operation triggers the reloading of all user data into memory.

Reviewer: ```suggestion
- When the number of SQL users in TiDB is very large (for example, more than 100,000), operations such as creating users, modifying passwords, and setting permissions might experience significant performance degradation. This occurs because each user update operation triggers the reloading of all user data into memory.
```

> - To optimize performance in environments with a large number of users, it is recommended to set the related variable to `ON`.

Reviewer: ```suggestion
- To optimize performance in environments with a large number of users, it is recommended to set this variable to `ON`.
```

- Default value: (system dependent)
- This variable shows the system time zone from when TiDB was first bootstrapped. See also [`time_zone`](#time_zone).

> ### `tidb_accelerate_user_creation_update` <span class="version-mark">New in v9.0.0</span>

Reviewer 2: ```suggestion
### tidb_accelerate_user_creation_update <span class="version-mark">New in v9.0.0</span>
```

- Scope: GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Boolean
- Default value: `OFF`

> - When too many (for example 100K) SQL users are created in the database, operations such as create user/modify password/change user privilege would suffer severe performance impact. The reason is that every modification to the user cause reload of all users' data. Setting the variable to `ON` is best practice is such scenario.

Reviewer 2: ```suggestion
- When too many (for example 100,000) SQL users are created in the database, subsequent operations such as user creation, password modification, and permission setting might experience significant performance degradation. This is because every user update requires reloading all user data. In this scenario, it is recommended to set this variable to ON.
```

    * `0` or `OFF`: modify user related operations cause reloading of all users' data, the performance is not good.

>     * `1` or `ON`: modify user related operations only reload the active users, so the performance is better.

Reviewer 2: ```suggestion
    * `1` or `ON`: modify user related operations only reload the active users, so the performance is better.
        - Modifying operations for users who are not logged in will not trigger the loading of all user data into memory.
        - Modifying operations for logged-in users will only reload the updated user data into memory.
        - This provides better performance in scenarios with a large number of users.
```

>     * `0` or `OFF`: modify user related operations cause reloading of all users' data, the performance is not good.
>     * `1` or `ON`: modify user related operations only reload the active users, so the performance is better.

Reviewer 3: ```suggestion
    * `0` or `OFF`: When performing operations such as creating users, changing passwords, or setting permissions, each operation triggers a full reload of all user data into memory. This may result in poor performance in scenarios with a large number of users.
    * `1` or `ON`: 
    Creating users or performing password changes and permission settings for non-logged-in users will not trigger a full reload of all user data into memory.
    Modifying logged-in users will only reload the updated user data into memory.
    This provides better performance in scenarios with a large number of users.
```

- Default value: (system dependent)
- This variable shows the system time zone from when TiDB was first bootstrapped. See also [`time_zone`](#time_zone).

### tidb_accelerate_user_creation_update <span class="version-mark">New in v9.0.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Boolean
- Default value: `OFF`
- When too many (for example 100,000) SQL users are created in the database, subsequent operations such as user creation, password modification, and permission setting might experience significant performance degradation. This is because every user update requires reloading all user data. In this scenario, it is recommended to set this variable to ON.
    * `0` or `OFF`: modify user related operations cause reloading of all users' data, the performance is not good.
    * `1` or `ON`: modify user related operations only reload the active users, so the performance is better.

>         - Modifying operations for users who are not logged in will not trigger the loading of all user data into memory.

Reviewer 2: ```suggestion
        - Creating users or performing operations such as modifying passwords and setting permissions for users who are not logged in will not trigger the loading of all user data into memory.
```

> - When too many (for example 100,000) SQL users are created in the database, subsequent operations such as user creation, password modification, and permission setting might experience significant performance degradation. This is because every user update requires reloading all user data. In this scenario, it is recommended to set this variable to ON.
>     * `0` or `OFF`: when performing operations such as creating users, modifying passwords, and setting permissions, each operation will trigger a reload of all user data into memory. In scenarios with a large number of users, this setting might lead to performance degradation.
>     * `1` or `ON`: modify user related operations only reload the active users, so the performance is better.
>         - Creating users or performing operations such as modifying passwords and setting permissions for users who are not logged in will not trigger the loading of all user data into memory.
>         - Modifying operations for logged-in users will only reload the updated user data into memory.
>         - This provides better performance in scenarios with a large number of users.

Reviewer 2: ```suggestion
- When the number of SQL users in the system is too large (for example 100,000), operations such as creating users, modifying passwords, and setting permissions might experience significant performance degradation. This is because each user update operation triggers the reloading of all user data into memory.
- To optimize performance, it is recommended to set the relevant variables to `ON` in scenarios with a large number of users.
- `0` or `OFF`: When set to `0` or `OFF`, every time a user creation, password modification, or permission setting operation is performed, TiDB will reload all user data into memory. In scenarios with a large number of users, this setting might cause significant performance degradation.
- `1` or `ON`: When set to `1` or `ON`, TiDB will use a more efficient user data loading strategy:
    - When creating users or performing operations such as modifying passwords and setting permissions for non-logged-in users, TiDB **will not** reload all user data into memory.
    - When performing operations on logged-in users, the system will only reload the updated data for those specific users into memory.
    - In scenarios with a large number of users, this setting can significantly improve performance.
```

- Default value: (system dependent)
- This variable shows the system time zone from when TiDB was first bootstrapped. See also [`time_zone`](#time_zone).

### tidb_accelerate_user_creation_update <span class="version-mark">New in v9.0.0</span>

- Scope: GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Boolean
- Default value: `OFF`
- When the number of SQL users in the system is too large (for example 100,000), operations such as creating users, modifying passwords, and setting permissions might experience significant performance degradation. This is because each user update operation triggers the reloading of all user data into memory.
- To optimize performance, it is recommended to set the relevant variables to `ON` in scenarios with a large number of users.

> - `OFF` or `0`: when set to ``OFF` or `0`, every time a user creation, password modification, or permission setting operation is performed, TiDB will reload all user data into memory. In scenarios with a large number of users, this setting might cause significant performance degradation.

Reviewer 2: ```suggestion
- `OFF` or `0`: when set to `OFF` or `0`, every time a user creation, password modification, or permission setting operation is performed, TiDB will reload all user data into memory. In scenarios with a large number of users, this setting might cause significant performance degradation.
```

> - `0` or `OFF`: when set to `0` or `OFF`, every time a user creation, password modification, or permission setting operation is performed, TiDB will reload all user data into memory. In scenarios with a large number of users, this setting might cause significant performance degradation.
> - `1` or `ON`: when set to `1` or `ON`, TiDB will use a more efficient user data loading strategy:

Reviewer 2: ```suggestion
- `OFF` or `0`: when set to ``OFF` or `0`, every time a user creation, password modification, or permission setting operation is performed, TiDB will reload all user data into memory. In scenarios with a large number of users, this setting might cause significant performance degradation.
- `ON` or `1`: when set to `ON` or `1`, TiDB will use a more efficient user data loading strategy:
```

> - `OFF` or `0`: when set to `OFF` or `0`, every time a user creation, password modification, or permission setting operation is performed, TiDB will reload all user data into memory. In scenarios with a large number of users, this setting might cause significant performance degradation.
> - `ON` or `1`: when set to `ON` or `1`, TiDB will use a more efficient user data loading strategy:

Reviewer 2: ```suggestion
- `OFF` or `0`: every time a user creation, password modification, or permission setting operation is performed, TiDB will reload all user data into memory. In scenarios with a large number of users, this setting might cause significant performance degradation.
- `ON` or `1`: TiDB uses a more efficient user data loading strategy as follows:
```

>     - When creating users or performing operations such as modifying passwords and setting permissions for non-logged-in users, TiDB **will not** reload all user data into memory.
>     - When performing operations on logged-in users, the system will only reload the updated data for those specific users into memory.

Reviewer 2: ```suggestion
    - When creating users or performing operations such as modifying passwords and setting permissions for **non-logged-in** users, TiDB **will not** reload all user data into memory.
    - When performing operations on **logged-in** users, the system will only reload the updated data for those specific users into memory.
```

>     - When creating users or performing operations such as modifying passwords and setting permissions for **non-logged-in** users, TiDB **will not** reload all user data into memory.
>     - When performing operations on **logged-in** users, the system will only reload the updated data for those specific users into memory.

Reviewer 2: ```suggestion
    - When creating users or performing operations such as modifying passwords and setting permissions for **non-logged-in** users, TiDB **does not** reload all user data into memory.
    - When performing operations on **logged-in** users, the system only reloads the updated data for those specific users into memory.
```

> - When the number of SQL users in the system is too large (for example 100,000), operations such as creating users, modifying passwords, and setting permissions might experience significant performance degradation. This is because each user update operation triggers the reloading of all user data into memory.
> - To optimize performance, it is recommended to set the relevant variables to `ON` in scenarios with a large number of users.
> - `OFF` or `0`: every time a user creation, password modification, or permission setting operation is performed, TiDB will reload all user data into memory. In scenarios with a large number of users, this setting might cause significant performance degradation.
> - `ON` or `1`: TiDB uses a more efficient user data loading strategy as follows:
>     - When creating users or performing operations such as modifying passwords and setting permissions for **non-logged-in** users, TiDB **does not** reload all user data into memory.
>     - When performing operations on **logged-in** users, the system only reloads the updated data for those specific users into memory.
>     - In scenarios with a large number of users, this setting can significantly improve performance.

Reviewer 2: ```suggestion
- When the number of SQL users in the system is very large (for example 100,000), operations such as creating users, modifying passwords, and setting permissions might experience significant performance degradation. This occurs because each user update operation triggers the reloading of all user data into memory.
- To optimize performance in environments with a large number of users, it is recommended to set the related variable to `ON`.
- `OFF` or `0`: every time an operation such as user creation, password modification, or permission setting is performed, TiDB reloads all user data into memory. In scenarios with many users, this setting might cause significant performance degradation.
- `ON` or `1`: TiDB uses a more efficient user data loading strategy as follows:
    - When you create users or perform operations such as modifying passwords and setting permissions for **non-logged-in** users, TiDB **does not** reload all user data into memory.
    - When you perform operations on **logged-in** users, TiDB only reloads the updated data for those specific users into memory.
    - In scenarios with many users, this setting can significantly improve performance.
```

- Default value: `ON`. Before v8.5.0, the default value is `OFF`.
- This variable is used to control whether to enable [TiDB Accelerated Table Creation](/accelerated-table-creation.md).
- Starting from v8.0.0, TiDB supports accelerating table creation by the [`CREATE TABLE`](/sql-statements/sql-statement-create-table.md) statement using `tidb_enable_fast_create_table`.

> - This variable is renamed from the variable [`tidb_ddl_version`](https://docs.pingcap.com/tidb/stable/system-variables#tidb_ddl_version-new-in-v760) that is introduced in v7.6.0. Starting from v8.0.0, `tidb_ddl_version` no longer takes effect.

Reviewer 2: ```suggestion
- This variable is renamed from the variable [`tidb_ddl_version`](https://docs-archive.pingcap.com/tidb/v7.6/system-variables#tidb_ddl_version-new-in-v760) that is introduced in v7.6.0. Starting from v8.0.0, `tidb_ddl_version` no longer takes effect.
```

- Default value: `""`
- This variable is used to control some internal behaviors of the optimizer.
- The optimizer's behavior might vary depending on user scenarios or SQL statements. This variable provides a more fine-grained control over the optimizer and helps to prevent performance regression after upgrading caused by behavior changes in the optimizer.

> - For a more detailed introduction, see [Optimizer Fix Controls](https://docs.pingcap.com/tidb/stable/optimizer-fix-controls).

Reviewer: ```suggestion
- For a more detailed introduction, see [Optimizer Fix Controls](/optimizer-fix-controls.md).
```

>
> - Depending on the specific business scenario, enabling this option might cause a certain degree of throughput reduction (average latency increase) for transactions with frequent lock conflicts.
> - This option only takes effect on statements that need to lock a single key. If a statement needs to lock multiple rows at the same time, this option will not take effect on such statements.

> > - This feature is introduced in v6.6.0 by the [`tidb_pessimistic_txn_aggressive_locking`](https://docs.pingcap.com/tidb/stable/system-variables#tidb_pessimistic_txn_aggressive_locking-new-in-v660) variable, which is disabled by default.

Reviewer 2: ```suggestion
> - This feature is introduced in v6.6.0 by the [`tidb_pessimistic_txn_aggressive_locking`](https://docs-archive.pingcap.com/tidb/v6.6/system-variables#tidb_pessimistic_txn_aggressive_locking-new-in-v660) variable, which is disabled by default.
```
